### PR TITLE
reporters/gerrit: provide build itself for summary callback

### PR DESCRIPTION
**Pull request by sa2ajj**
Summary callback differs from the build review callback in lack of the
actual build instance: the former gets only particular information about
builds.

In some situations, the build itself is required to do some magic.

(Ported from eight)

### CODE REVIEW

File: master/buildbot/reporters/gerrit.py [hunk 1]
```diff
@@ -329,7 +329,8 @@ def getBuildInfo(build):
                         'result': result,
                         'resultText': resultText,
                         'text': build['state_string'],
-                        'url': utils.getURLForBuild(self.master, build['builder']['builderid'], build['number'])
+                        'url': utils.getURLForBuild(self.master, build['builder']['builderid'], build['number']),
```
Comment (sa2ajj) [inline master/buildbot/reporters/gerrit.py]: @tardyp should I use new `getUrl` method of the build here?

**Comment by sa2ajj** [comment]
transient failure?

File: master/buildbot/reporters/gerrit.py (hunk 1 above)
Comment (djmitche) [inline master/buildbot/reporters/gerrit.py]: `build` is a dictionary, right?

**Comment by djmitche** [comment]
This looks fine to me, but reporters are not my area of expertise

File: master/buildbot/reporters/gerrit.py (hunk 1 above)
Comment (sa2ajj) [inline master/buildbot/reporters/gerrit.py]: oh, yes, at this point it's a dictionary...

(I need to dig a bit more...)

Comment (tardyp) [inline master/buildbot/reporters/gerrit.py]: build.getUrl is from process.build.Build class, which is the running build object which you can get in your custom steps.

in reporters, you are bound to the data api.

**Comment by tardyp** [comment]
yes, I have introduced an instable test recently :-}

I need to fix this
buildbot.test.integration.test_URLs.UrlForBuildMaster.test_transfer

**Pull request merged**